lib/components/base-components/NormalComponent/NormalComponent.ts

```diff
@@ -375,6 +376,22 @@ export class NormalComponent<
     return s.startsWith("http://") || s.startsWith("https://")
   }
 
+  /**
+   * Returns the library prefix and footprintName̦
+   * reference - "<lib>:<footprintName>"
+   */
+  _parseLibraryFootprintRef(
+    s: string,
+  ): { lib: string; footprintName: string } | null {
+    if (this._isFootprintUrl(s)) return null
+    const idx = s.indexOf(":")
+    if (idx <= 0) return null
+    const lib = s.slice(0, idx)
+    const footprintName = s.slice(idx + 1)
+    if (!lib || !footprintName) return null
+    return { lib, footprintName }
```

Comment:
i think this breaks url handling ("http" or "https" would be the lib), i would also qualify lib because it's a better variable name and also done for footprint
 
```suggestion
    return { footprintLib, footprintName }
```